scripts/dts/edtlib.py

```diff
@@ -984,6 +996,43 @@ def _init_binding(self):
         # No binding found
         self._binding = self.binding_path = self.matching_compat = None
 
+    def _binding_from_properties(self):
+        # Returns a binding synthesized from the properties in the node.
+
+        if self.compats:
+            _err(f"compatible in node with inferred binding: {self.path}")
+
+        self._binding = OrderedDict()
+        self.matching_compat = self.path.split('/')[-1]
+        self.compats = [self.matching_compat]
+        self.binding_path = "@synthesized/" + self.matching_compat
```

Comment:
Is this necessary, or can it be set to None? I'm not fond of faking out values like this.

Reply:
As implemented this produces:
```
 * Devicetree node: /zephyr,user
 *
 * Node's generated path identifier: DT_N_S_zephyr_user
 *
 * Binding (compatible = zephyr,user):
 *   @synthesized/zephyr,user
 *
 * Dependency Ordinal: 9
 *
 * Requires:
 *   0   /
 *   7   /soc/gpio@50000000
 *   8   /soc/gpio@50000300
 */
```

Setting to `None` produces:

```
Traceback (most recent call last):
  File "/mnt/nordic/zp/zephyr/scripts/dts/gen_defines.py", line 744, in <module>
    main()
  File "/mnt/nordic/zp/zephyr/scripts/dts/gen_defines.py", line 82, in main
    write_node_comment(node)
  File "/mnt/nordic/zp/zephyr/scripts/dts/gen_defines.py", line 184, in write_node_comment
    {relativize(node.binding_path)}
  File "/mnt/nordic/zp/zephyr/scripts/dts/gen_defines.py", line 219, in relativize
    return str("$ZEPHYR_BASE" / pathlib.Path(path).relative_to(zbase))
  File "/usr/lib/python3.8/pathlib.py", line 1033, in __new__
    self = cls._from_parts(args, init=False)
  File "/usr/lib/python3.8/pathlib.py", line 674, in _from_parts
    drv, root, parts = self._parse_args(args)
  File "/usr/lib/python3.8/pathlib.py", line 658, in _parse_args
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

That can be changed if you wish; please provide the text you'd like to see for a synthesized binding.

Reply:
After some dithering:
```
 * Binding (compatible = nordic,nrf-gpio):
 *   $ZEPHYR_BASE/dts/bindings/gpio/nordic,nrf-gpio.yaml
...
 * Binding (compatible = zephyr,user):
 *   No yaml (bindings inferred from properties)
```
OK?